test/functional/Java22andUp/src/org/openj9/test/jep454/downcall/UnionStructTests.java

```diff
@@ -0,0 +1,946 @@
+		MemorySegment functionSymbol = nativeLibLookup.find("addDoubleAndDoublesFromStruct_Nested2DoubleUnion").get();
+		MethodHandle mh = linker.downcallHandle(functionSymbol, fd);
+
+		try (Arena arena = Arena.ofConfined()) {
+			MemorySegment structSegmt = arena.allocate(structLayout);
+			structSegmt.set(JAVA_DOUBLE, 0, 2228.111D);
+			structSegmt.set(JAVA_DOUBLE, 8, 2229.221D);
+
+			double result = (double)mh.invokeExact(3336.333D, structSegmt);
+			Assert.assertEquals(result, 10022.886D, 0.001D);
+		}
+	}
+
+	@Test
+	public void test_addDoubleAndDoublesFromUnion_Nested2DoubleStruct() throws Throwable {
+		GroupLayout nestedStructLayout = MemoryLayout.structLayout(JAVA_DOUBLE.withName("elem1"), JAVA_DOUBLE.withName("elem2"));
+		GroupLayout unionLayout = MemoryLayout.unionLayout(JAVA_DOUBLE.withName("elem1"), nestedStructLayout.withName("struct_elem2"));
+		FunctionDescriptor fd = FunctionDescriptor.of(JAVA_DOUBLE, JAVA_DOUBLE, unionLayout);
+		MemorySegment functionSymbol = nativeLibLookup.find("addDoubleAndDoublesFromUnion_Nested2DoubleStruct").get();
+		MethodHandle mh = linker.downcallHandle(functionSymbol, fd);
+
+		try (Arena arena = Arena.ofConfined()) {
+			MemorySegment unionSegmt = arena.allocate(unionLayout);
+			unionSegmt.set(JAVA_DOUBLE, 0, 2228.111D);
+			unionSegmt.set(JAVA_DOUBLE, 8, 2229.221D);
+
+			double result = (double)mh.invokeExact(3336.333D, unionSegmt);
+			Assert.assertEquals(result, 10021.776D, 0.001D);
+		}
+	}
+
+	@Test
+	public void test_addDoubleAndDoublesFromUnion_Nested4DoubleStruct() throws Throwable {
+		GroupLayout nestedStructLayout = MemoryLayout.structLayout(JAVA_DOUBLE.withName("elem1"),
+		JAVA_DOUBLE.withName("elem2"), JAVA_DOUBLE.withName("elem3"), JAVA_DOUBLE.withName("elem4"));
+		GroupLayout unionLayout = MemoryLayout.unionLayout(JAVA_DOUBLE.withName("elem1"), nestedStructLayout.withName("struct_elem2"));
+		FunctionDescriptor fd = FunctionDescriptor.of(JAVA_DOUBLE, JAVA_DOUBLE, unionLayout);
+		MemorySegment functionSymbol = nativeLibLookup.find("addDoubleAndDoublesFromUnion_Nested4DoubleStruct").get();
+		MethodHandle mh = linker.downcallHandle(functionSymbol, fd);
+
+		try (Arena arena = Arena.ofConfined()) {
+			MemorySegment unionSegmt = arena.allocate(unionLayout);
+			unionSegmt.set(JAVA_DOUBLE, 0, 2228.111D);
+			unionSegmt.set(JAVA_DOUBLE, 8, 2229.221D);
+			unionSegmt.set(JAVA_DOUBLE, 16, 2230.221D);
+			unionSegmt.set(JAVA_DOUBLE, 24, 2231.221D);
+
+			double result = (double)mh.invokeExact(3336.333D, unionSegmt);
+			Assert.assertEquals(result, 14483.218D, 0.001D);
+		}
+	}
+
+	@Test
+	public void test_add2DoubleStructs_returnStruct_Nested2DoubleUnion() throws Throwable {
+		GroupLayout structLayout = MemoryLayout.structLayout(JAVA_DOUBLE.withName("elem1"), JAVA_DOUBLE.withName("elem2"));
+		VarHandle doubleHandle1 = structLayout.varHandle(PathElement.groupElement("elem1"));
+		VarHandle doubleHandle2 = structLayout.varHandle(PathElement.groupElement("elem2"));
```

Comment:
Please remove these unused locals and the import of `VarHandle`, and do the same for 1Java21Only`.

Reply:
Fixed.